### DISCUSSION TITLE
`LiteClient.getTransactions` throws an error (supposedly) for old addresses

### DISCUSSION THREAD

**Discussion by mcnckc** [andreypfau/ton-kotlin#162] [Q&A] [answered]
I am using ton-kotlin 0.3.1 installed like that
```
implementation 'org.ton:ton-kotlin-tvm:0.3.1'
implementation 'org.ton:ton-kotlin-block-tlb:0.3.1'
implementation 'org.ton:ton-kotlin-liteclient:0.3.1'
```

Here is a toy example of using this function

```
val address = "UQBZf6LoZG0KSONnGs1ljXgVe_KhxOspnT1NAwV0rWMy1mw9"
val liteClient = LiteClient(Dispatchers.IO, config)
runBlocking {
    val result = liteClient.getAccountState(AddrStd(address))
    liteClient.getTransactions(result.address, result.lastTransactionId!!, 2)
}
```

For addresses 
-https://dev.tonviewer.com/EQBZf6LoZG0KSONnGs1ljXgVe_KhxOspnT1NAwV0rWMy1jH4
-https://dev.tonviewer.com/EQBWl0HakAranxsLaDPcQGJtExFYZ8g7NR4D-JdvaV4cAKfZ

`getTransactions` throws following error:
```
Caused by: org.ton.lite.api.exception.LiteServerUnknownException: cannot locate transaction in block with specified logical time
```

For address
-https://dev.tonviewer.com/EQDaokYMTQSpFFjbe9Pht2AH6AwD2PhnOGmSFsJH5WsNhVtp

It throws:
```
Caused by: org.ton.lite.api.exception.LiteServerUnknownException: cannot load block (0,8000000000000000,42712115):B715FD8023F2E2430ED88D4A112FE58F80A59DECCFA09AAE4F35EC2F2A9CA599:5016E35CC8264C8DD98925C40753D42C1EFFE598B7CC2BA06D41B2F142E08DF3 with specified transaction: not in db
```

However for addresses
-https://dev.tonviewer.com/EQCtiv7PrMJImWiF2L5oJCgPnzp-VML2CAt5cbn1VsKAxLiE
-https://dev.tonviewer.com/EQCTd6HSN6kMQfYU5fel7BykpGh7ARzWunj2zmB9fhgGA2Y6

It works without exceptions

It seems, that "bad" addresses only differ in last transaction date.

What could be the reason for such behaviour, and can it be fixed?

**Comment by andreypfau** [comment, marked as answer]
It's not issue of ton-kotlin.
This happens because you are requesting data from a light server that is not an archive node, and since it does not have data for the selected period, it returns the error "not in db"
